### PR TITLE
Buddy Pokemon task

### PR DESCRIPTION
**Pull request by DeXtroTip**
## Short Description:
Adds the new buddy system feature to the bot.
Able to select by family and automatically switch buddy at candy limit using a list of pokemon to use.

## Fixes/Resolves/Closes (please use correct syntax):
- #5456 
- #5419

### CODE REVIEW

**Comment by solderzzc** [comment]
@DeXtroTip I don't see any 0.37.0 related code on the pgodevapi. how do you think about?
https://github.com/pogodevorg/pgoapi/commits/master

**Comment by DeXtroTip** [comment]
@solderzzc I think the api hasn't really changed with the 0.37.0 update and the calls for the buddy system were introduced before. So, it works, I just don't really know if it's safe or not.

**Comment by solderzzc** [comment]
@DeXtroTip  sure, let's have a test after merge.

**Comment by solderzzc** [comment]
👍

**Comment by alexyaoyang** [comment]
👍

**Comment by Gobberwart** [comment]
Looks good. Go ahead and merge.

**Comment by davidakachaos** [comment]
@DeXtroTip could you solve the conflicts perhaps?

**Comment by DeXtroTip** [comment]
I'm not home right now, I'll do it later today, as soon as I can.

**Comment by davidakachaos** [comment]
That's great! Thanks!

Op 20 sep. 2016 13:28 schreef "DeXtroTip" <notifications@github.com>:

File: configs/config.json.cluster.example [hunk 1]
```diff
@@ -39,6 +39,21 @@
...
+        "config": {
+          "enabled": true,
+          "buddy_list": "dratini, magikarp",
+          "best_in_family": true,
+          "// candy_limit = 0 means no limit, so it will never change current buddy": {},
+          "candy_limit": 0,
+          "// force_first_change = true will always change buddy at start removing current one": {},
+          "force_first_change": false,
+          "buddy_change_wait_min": 3,
+          "buddy_change_wait_min": 5,
```
Comment (davidakachaos) [inline configs/config.json.cluster.example]: * buddy_change_wait_max

**Comment by DeXtroTip** [comment]
Fixed all conflicts. Need new approve before merge.

**Comment by alexyaoyang** [comment]
👍 Looks good!

**Pull request merged**